documentation/localization_howto.md

```diff
@@ -0,0 +1,165 @@
+
+Filename conventions
+--------------------
+
+In order to differentiate a localized file, append the lowercase language code after the
+filename with a 'dash' symbol (e.x. README-fr.md for a README localized in French).
+
+Point of references
+-------------------
+
+When translating, English should be considered the point of reference to translate *from*.
+
+READMEs and Documentation
+-------------------------
+
+The first interaction with Streisand for an end user would be first visiting the README,
+where all the necessary steps one would need to set up their computing environment are
+detailed, as well as a listing of all the VPN services provided by Streisand are listed.
+
+Unlike Streisand's instructions markdown template files, READMEs and documentation are
+orthogonal to Streisand being able to successfully build. As a consequences, it is possible
+to have READMEs and documentation in more languages than Streisand builds with.
+
+Defining locales
+----------------
+
+In order to define a locale for Streisand, a new entry must defining your language must be
+added to the `streisand_languages` dict object in the `playbooks/common/vars/main.yml` file,
+where the 2 letter code is considered the key, and a list of key-value pairs with additional data
+as its value.
+
+For example:
+
+```
+streisand_languages:
+  en:
+    file_suffix: ""
+    language_name: "English"
+    tor_locale: "en-US"
+  fr:
+    file_suffix: "-fr"              # appends to the end of a file name
+    language_name: "Français"       # name of the language in the language itself
+    tor_locale: "fr"                # edge case for Tor          
+```
+
+Localizing gateway instructions
+-------------------------------
+
+The following roles contain user instructions that will require translation:
+
+ - `playbooks/roles/l2tp-ipsec/templates/`
+  1. `instructions.md.j2`
+
+ - `playbooks/roles/openconnect/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/openvpn/templates`
+  1. `instructions.md.j2`
+  1. `stunnel-insructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/shadowsocks/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+- `playbooks/roles/ssh-forward/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/streisand-gateway/templates`
+  1. `index.md.j2`
+  1. `instructions.md.j2`
+  1. `firewall.md.j2`
+
+ - `playbooks/roles/streisand-mirror/templates`
+  1. `mirror-index.md.j2`
+
+ - `playbooks/roles/stunnel/templates`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/tor/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/wireguard/templates`
+  1. `instructions.md.j2`
+
+Language selection in generated pages
+-------------------------------------
+
+By default, Streisand's common header files contains language selection 
+(generated using the `streisand_languages` dict), and for most pages will be done
```

Comment:
Maybe split this up to a new sentence. "For most pages the header with language selection links will be included automatically"